crates/utils/src/tracing/grpc.rs

```diff
@@ -1,3 +1,31 @@
+use std::{fmt::Display, time::Duration};
+
+/// A [`tower_http::trace::OnFailure`] implementation which creates events of level dependant on the
+/// failure classification.
+#[derive(Clone)]
+pub struct OnFailure {}
+
+impl<F: Display> tower_http::trace::OnFailure<F> for OnFailure {
+    fn on_failure(&mut self, failure_class: F, latency: Duration, _span: &tracing::Span) {
+        let latency = format!("{} ms", latency.as_millis());
+        if failure_class.to_string().as_str() == "Code: 12" {
+            // Unimplemented failure classifications occur for example when
+            // external load balancers send healthchecks (expecting status code 12).
```

Comment:
Are you saying the load-balancer expects to receive an error as part of its health check?

Reply:
Ah I see [here](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-health-checks.html). Should we not be changing these from the AWS defaults instead?

So that it hits a gRPC method that exists, and we change the code to `0` or whatever gRPC indicates as success?

Reply:
The default for gRPC is `/AWS.ALB/healthcheck` which of course fails with error code 12 because it is indeed unimplemented.

Should we change that to `/rpc.Api/Status` (or whatever that is), and then fix the code as well?

Reply:
Ah thanks I didn't have the full path correct. Hopefully that works. Let you know ASAP

Reply:
That fixed it thanks. Will close this.